app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryEffectHandler.kt

```diff
@@ -54,6 +55,26 @@ class CustomDrugEntryEffectHandler @AssistedInject constructor(
     return listOfNotNull(name, dosage, frequency?.toString()).joinToString()
   }
 
+  private fun updatePrescription(): ObservableTransformer<UpdatePrescription, CustomDrugEntryEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .flatMap { effect ->
+            prescriptionRepository.softDeletePrescription(effect.prescribedDrugUuid)
```

Comment:
Yes, we are just setting the `isDeleted` value to `true` here

Reply:
That is not what I am saying. The `softDeletePrescription` method returns a `Completable` that is never getting subscribed to. Hence, the actual database call will never be made and tests will continue to pass because the thing we verify in the tests is that the repository call is being made, not that the completable it returns is being subscribed to.

Reply:
So the `andThen` method Returns an Observable which will subscribe to this Completable and once that is completed then will subscribe to the next ObservableSource. An error event from this Completable will be propagated to the downstream subscriber and will result in skipping the subscription of the Observable. and it throws a `NullPointerException` when next is `null`. 

Therefore we are subscribing to it already. Correct me if I'm wrong tho.

This is also how it is being handled everywhere else as well(You can maybe look at `DosagePickerEffectHandler#deleteExistingPrescription` or `ConfirmRemovePrescriptionDialogEffectHandler#removePrescription`).

Reply:
The only thing the `andThen` method is subscribing to is an `Observable` that emits `CustomDrugSaved`. `PrescriptionRepository#softDeletePrescription` returns a `Completable` that is getting ignored.

Reply:
@vinaysshenoy Can you please suggest what should be done here instead? 😅

Reply:
Can you look at existing usages of the repository method and see what is happening differently at other usage sites?

Reply:
:face_palm:

Reply:
The problem was I was looking at `CustomDrugEntryEffectHandler#removePrescription` method all this while.

Reply:
😅

Reply:
@janhavisinghh Have you pushed the changes? I don't see any change in the method implementation.

Reply:
I added `.observeOn(schedulersProvider.io())` in this method.. Can you check again?

Reply:
That's not what I was referring to at all. I was referring to the return value of `PrescriptionRepository#softDeletePrescription` being unused.

Reply:
I've been looking at all the other implementations of this. And I don't see any difference. Am I going crazy? 😅 I looked at how `Completable` return types are being handled as well. I really don't see any difference. 
For ex:  
this is the `ConfirmRemovePrescriptionDialogEffectHandler#removePrescription` method : 
```
private fun removePrescription(): ObservableTransformer<RemovePrescription, ConfirmRemovePrescriptionDialogEvent> {
    return ObservableTransformer { effects ->
      effects
          .observeOn(schedulersProvider.io())
          .flatMap {
            prescriptionRepository.softDeletePrescription(it.prescriptionUuid)
                .andThen(Observable.just(PrescriptionRemoved))
          }
    }
  }
```

Or `DosagePickerEffectHandler#removePrescription`

```
private fun deleteExistingPrescription(): ObservableTransformer<DeleteExistingPrescription, DosagePickerEvent> {
    return ObservableTransformer { effects ->
      effects
          .observeOn(schedulers.io())
          .map { it.prescriptionUuid }
          .flatMap {
            prescriptionRepository
                .softDeletePrescription(it)
                .andThen(Observable.just(ExistingPrescriptionDeleted as DosagePickerEvent))
          }
    }
  }
```

Reply:
Your current implementation does not do anything with the `Completable` returned from `PrescriptionRepository#softDeletePrescription`. This means that the completable will only be created, but the actual database operation will never be executed because that completable is never subscribed to.

In both the examples you have shared, the `Completable` that is returned is chained with the next operation to be done using the `andThen` operator, which is not being done in your current implementation.

Reply:
urgh. :tears: done.

Reply:
Don't sweat it, we all have those days. 😅